scripts/cybershake/cybershake_progress.py

```diff
@@ -0,0 +1,320 @@
+from typing import List, Iterable
+import sys
+
+import numpy as np
+import pandas as pd
+from urllib.request import urlopen
+
+
+from estimation.estimate_cybershake import main as est_cybershake
+from scripts.management.MgmtDB import MgmtDB
+import qcore.simulation_structure as sim_struct
+import qcore.constants as const
+from qcore.formats import load_fault_selection_file
+from qcore.utils import load_yaml
+
+# The process types that are used for progress tracking
+PROCESS_TYPES = [
+    const.ProcessType.EMOD3D.str_value,
+    const.ProcessType.HF.str_value,
+    const.ProcessType.BB.str_value,
+]
+
+EST_CORE_HOURS_COL = "est_core_hours"
+ACT_CORE_HOURS_COL = "act_core_hours"
+
+R_COUNT_COL = "r_count"
+
+SLACK_ALERT_CONFIG = "slack_alert.yaml"
+
+_fault_template_fmt = "{:<15}{:<22}{:<7}{:<3}{:<17}{:<7}{:<3}{:<17}{:<7}{:<3}{:<22}"
+_summary_template_fmt = "{:<6} : {:>5}/{:>5} ({:4.2f}%) RELs complete: {:10.2f} CH consumed / {:10.2f} CH est ({: >#06.2f}%) {:10.2f} CH remaining est\n"
+
+
+def get_chours(json_file, proc_types: List[str], debug=False):
+    """Gets the core hours for the specified process types (i.e. EMOD3D, HF, BB, ...)
+    for the specified simulation metadata json log file.
+    """
+    with open(json_file, "r") as f:
+        data = json.load(f)
+    sim_name = data.get(const.MetadataField.sim_name.value)
+    core_hours = []
+    completed_proc = {}
+    for proc_type in proc_types:
+        if proc_type in data.keys():
+            core_hours.append(
+                data[proc_type].get(const.MetadataField.run_time.value, np.nan)
+                * data[proc_type].get(const.MetadataField.n_cores.value, np.nan)
+                / 3600.0
+            )
+
+        else:
+            core_hours.append(np.nan)
+    return sim_name, core_hours
+
+
+def get_chours_used(fault_dirs: Iterable[str]):
+    """Returns a dataframe containing the core hours used for each of the
+    specified faults"""
+    faults, core_hours, completed_r_counts, missing_data = [], [], [], []
+    for fault in fault_dirs:
+        fault = Path(fault)
+        fault_name = fault.stem
+        cur_fault_chours = np.asarray(
+            [
+                get_chours(json_file, PROCESS_TYPES)[1]
+                for json_file in fault.glob(f"**/{const.METADATA_LOG_FILENAME}")
+            ]
+        ).reshape(-1, len(PROCESS_TYPES))
+
+        faults.append(fault_name)
+        if cur_fault_chours.shape[0] > 0:
+            core_hours.append(np.nansum(cur_fault_chours, axis=0))
+        else:
+            core_hours.append([np.nan, np.nan, np.nan])
+
+    df = pd.DataFrame(index=faults, data=core_hours, columns=PROCESS_TYPES)
+
+    return df
+
+
+def get_faults_and_r_count(cybershake_list: str):
+    """Gets the fault names and number of realisations from a cybershake fault list."""
+
+    faults_dict = load_fault_selection_file(cybershake_list)
+    return np.asarray(list(faults_dict.keys())), np.asarray(list(faults_dict.values()))
```

Comment:
Can we trust this to maintain order between the two groups (keys  vs values)?

Reply:
Considering that the list of faults and counts are passed around together, I think that it would be a better idea to keep them together until the printing. 

I also don't think that sorting it is necessary. The order that the list.txt is in should be fine for progress checking.